### ISSUE TITLE
Early Stopping for Deep Learning Models

### ISSUE THREAD

**Issue by Cgarg9** [PEC-CSS/AI#43]
### Description:
Implement Early Stopping to improve training efficiency for a Deep Neural Network (DNN).

### Tasks:

- Train a DNN on the CIFAR-10 dataset without early stopping.
- Implement Early Stopping with patience and monitor validation loss.
- Compare training time and final accuracy with and without early stopping.
- Visualize loss curves before and after applying Early Stopping.
- Name the notebook early_stopping_dnn.ipynb.
- Update the README file with relevant references.

**Comment by Amisha2700** [comment]
@Cgarg9 can this issue be assigned to me

**Comment by Cgarg9** [comment]
sorry must have missed this. please dm me just in case when you ask to get assigned or raise a pr

**Comment by Cgarg9** [comment]
@Amisha2700 any progress?

**Comment by Amisha2700** [comment]
Yes sir. It's almost done.
Will make pr by tomorrow
